.gitlab-ci.yml

```diff
@@ -1,8 +1,22 @@
+before_script:
+  - >
+    if echo "$CI_COMMIT_TITLE" | grep -q "ci: Update snapshots"; then
+      echo "Skipping tests due to commit title"
+      exit 0
+    fi
+
+workflow:
+  auto_cancel:
+    on_new_commit: interruptible
+    on_job_failure: all
```

Comment:
I'm not sure on this one. Technically it would be good that we catch more than one error, so that we can push a new commit that fixes them all, rather than finding out one by one.

Reply:
Yeah, I was also unsure about this flag. The reason I decided to introduce it is that normally the tests should be executed locally. The CI step should only validate the final state of a PR.

I don't have a strong opinion on this. If you prefer, I can revert it.

Reply:
I think in case the PR changes hands for some reasons, it's good to have a full picture of what the current state is, rather than relying on local-only context.

Reply:
Also, seeing how the integration tests are failing often, letting all other tests running could be good to then say "let's merge anyway since everything else works"